### PR TITLE
Feature/six point radial distortion homography

### PR DESCRIPTION
**Pull request by urbste**
This PR adds the six point radial distortion Homography solver proposed in:
Radial Distortion Homography, Zuzana Kukelova et al. CVPR 2015

The original Matlab / C++ implementation is taken from the authors: http://cmp.felk.cvut.cz/~hellej1/radhomo/radhomo-0.1.zip

In the paper two solvers are proposed: H5_l1l2 and H6_l1l2.
This PR implements the second solver, which is faster than the five point solver.

### CODE REVIEW

File: src/theia/sfm/pose/six_point_radial_distortion_homography.cc [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+using Eigen::Vector3d;
+using Eigen::Vector2d;
+using Vector6d = Eigen::Matrix<double, 6, 1>;
+using Array51d = Eigen::Array<double, 5, 1>;
+using Matrix68d = Eigen::Matrix<double, 6, 8>;
+using Matrix62d = Eigen::Matrix<double, 6, 2>;
+using Matrix65d = Eigen::Matrix<double, 6, 5>;
+using Eigen::Matrix3d;
+
+bool isNearZero(double val) {
```
Comment (sweeneychris) [inline src/theia/sfm/pose/six_point_radial_distortion_homography.cc]:
```suggestion
bool IsNearZero(double val) {
```

File: src/theia/sfm/pose/six_point_radial_distortion_homography.cc [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  return (+val < (100.0 * std::numeric_limits<double>::epsilon())) &&
+         (-val < (100.0 * std::numeric_limits<double>::epsilon()));
+}
+
+bool SixPointRadialDistortionHomography(
+    const std::vector<Eigen::Vector2d> &normalized_feature_points_left,
```
Comment (sweeneychris) [inline src/theia/sfm/pose/six_point_radial_distortion_homography.cc]:
```suggestion
    const std::vector<Eigen::Vector2d>& normalized_feature_points_left,
```

File: src/theia/sfm/pose/six_point_radial_distortion_homography.cc [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    const std::vector<Eigen::Vector2d> &normalized_feature_points_right,
```
Comment (sweeneychris) [inline src/theia/sfm/pose/six_point_radial_distortion_homography.cc]:
```suggestion
    const std::vector<Eigen::Vector2d>& normalized_feature_points_right,
```

File: src/theia/sfm/pose/six_point_radial_distortion_homography.cc [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    std::vector<RadialHomographyResult> *results, const double lmin,
```
Comment (sweeneychris) [inline src/theia/sfm/pose/six_point_radial_distortion_homography.cc]:
```suggestion
    std::vector<RadialHomographyResult>* results, const double lmin,
```

File: src/theia/sfm/pose/six_point_radial_distortion_homography.cc [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  M.col(2) = -X.col(1);
+  M.col(3) = X.col(0).array() * U.col(0).array();
+  M.col(4) = X.col(0).array() * U.col(1).array();
+  M.col(5) = X.col(0);
+  M.col(6) = -X.col(1).array() * u2.array();
+  M.col(7) = X.col(0).array() * u2.array();
+
+  Eigen::JacobiSVD<Matrix68d, Eigen::FullPivHouseholderQRPreconditioner> Svd1(
+      M, Eigen::ComputeFullV);
+  const Eigen::Matrix<double, 8, 8> &V1 = Svd1.matrixV();
```
Comment (sweeneychris) [inline src/theia/sfm/pose/six_point_radial_distortion_homography.cc]:
```suggestion
  const Eigen::Matrix<double, 8, 8>& V1 = Svd1.matrixV();
```

File: src/theia/sfm/pose/six_point_radial_distortion_homography.cc (hunk 1 above)
Comment (urbste) [reply src/theia/sfm/pose/six_point_radial_distortion_homography.cc]: I am using qtcreator and the Beautifier with clang to format the files. In the preferences I can choose between styles: LLVM, Google, WebKit and some others. Which one do you use?

**Pull request merged**